### PR TITLE
Add some missing validations to the Firewall Application and Nat rule commands

### PR DESCRIPTION
**Pull request by tejasshah7**
## Description

Add some missing validations to the Firewall Application and Nat rule commands

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs [hunk 1]
```diff
@@ -70,6 +70,12 @@ public override void Execute()
         {
             base.Execute();
 
+            // One of SourceAddress or SourceIpGroup must be present
```
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs:73]: if one of these two parameter is required, it need to be handled by parameter sets:

one with SourceAddress  mandatory and the other with SourceIpGroup  mandatory.

File: src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs [hunk 2]
```diff
@@ -79,6 +79,17 @@ public override void Execute()
         {
             base.Execute();
 
+            // One of SourceAddress or SourceIpGroup must be present
```
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs:82]: if one of these two parameter is required, it need to be handled by parameter sets:

one with SourceAddress mandatory and the other with SourceIpGroup mandatory.

File: src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs [hunk 3]
```diff
@@ -79,6 +79,17 @@ public override void Execute()
...
+            if ((SourceAddress == null) && (SourceIpGroup == null))
```
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs:83]: if one of these two parameter is required, it need to be handled by parameter sets:

one with SourceAddress mandatory and the other with SourceIpGroup mandatory.

File: src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs [hunk 4]
```diff
@@ -79,6 +79,17 @@ public override void Execute()
...
+            {
+                throw new ArgumentException("Either SourceAddress or SourceIpGroup is required.");
+            }
+
+            if (DestinationAddress == null)
+            {
```
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs:89]: if it's required, please make the parameter mandatory

Comment (tejasshah7) [reply src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs]: Makes sense. Actually it is already mandatory so removed this check.

File: src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs (hunk 3 above)
Comment (tejasshah7) [reply src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs:83]: @VeryEarly 
I have not added parameter sets before. How exactly does it work? Should I just create a class for Parameter sets, added 2 contant strings, one for each and then specific corresponding ones for each of these fields? I dont understand how would that ensure at least one of the 2 fields is always present.

Comment (VeryEarly) [reply src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs:83]: let's say you have two parameter sets only, parameters provided must be fulfill one of the parameter set.

Make SourceAddress  mandatory for one set and SourceIpGroup mandatory mandatory for the other.

File: src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs [hunk 5]
```diff
@@ -70,6 +70,12 @@ public override void Execute()
...
+            if ((SourceAddress == null) && (SourceIpGroup == null))
```
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs:74]: Also create two parameter sets for this condition

File: src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs (hunk 3 above)
Comment (tejasshah7) [reply src/Network/Network/AzureFirewallPolicy/NatRule/NewAzureFirewallPolicyNatRuleCommand.cs]: Done

File: src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs (hunk 5 above)
Comment (tejasshah7) [reply src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs:74]: Done

File: src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs (hunk 1 above)
Comment (VeryEarly) [inline src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs:73]: can you also add same parameters for this condition?

Comment (tejasshah7) [reply src/Network/Network/AzureFirewallPolicy/ApplicationRuleCondition/NewAzureFirewallPolicyApplicationRuleCommand.cs:83]: Done

**Comment by VeryEarly** [comment]
merged in #12173

**Pull request closed without merging**